### ISSUE TITLE
Run Language model on the entire dataset. 

### ISSUE THREAD

**Issue by tanzir5** [odissei-lifecourse/life-sequencing-dutch#14]
Currently the LM is being pretrained on a subset of the training data (5% I think). There are two bottlenecks:
1. Creating the masked data before pretraining (requires CPU only). The Fat Rome node is good for this. Also I can do chunking to ensure we create masked data for the entire dataset. 
2. Loading the masked data and training using gpus. 
        - the gpu only gets one batch at a time. so gpus are fine. 
        - loading the entire data may not be feasible. we may need to use something like DASK library or see how chunking can work without significantly effecting efficiency due to huge I/O time

**Comment by tanzir5** [comment]
We have devised a somewhat different strategy involving a customIterableDataset where we use a generator to yield data from the disk and use HDF5 file for efficiency. More details in  #24  

The customIterableDataset is working as expected, but the pipeline exhausts all memory on 480 GB GCN due to parallel processing and the process gets killed. I need to do some trial-and-error on Snellius to figure out the right balance between memory and time efficiency (i.e., # of processes and the chunk_size to run in parallel)

**Comment by f-hafner** [comment]
closing; the parallel processing for the pipeline is in #37 .
